wallet/wallet-controller/src/lib.rs

```diff
@@ -175,11 +187,82 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static> Controller<T> {
             .map_err(ControllerError::NodeCallError)
     }
 
+    pub fn create_wallet(
+        &mut self,
+        file_path: impl AsRef<Path>,
+        mnemonic: mnemonic::Mnemonic,
+        passphrase: Option<&str>,
+    ) -> Result<(), ControllerError<T>> {
+        utils::ensure!(
+            !file_path.as_ref().exists(),
+            ControllerError::WalletFileError(
+                file_path.as_ref().to_owned(),
+                "File already exists".to_owned()
+            )
+        );
+
+        let db = wallet::wallet::open_or_create_wallet_file(file_path)
```

Comment:
I would prefer to have a distinction between open and create so that errors can't happen. 

Of course this depends on the context. but then if you're calling the funciton "create_wallet", then it shouldn't be possible to overwrite a wallet.